nf_core/pipelines/download.py

```diff
@@ -1361,8 +1389,8 @@ def singularity_download_image(
             log.debug(f"Deleting incompleted singularity image download:\n'{output_path_tmp}'")
             if output_path_tmp and os.path.exists(output_path_tmp):
```

Comment:
This is the cleanup function I mentioned. Now, it is not sufficient anymore, because the cache is not considered.